Added Preact

jrson83 opened the pull request:
I tested configuration with `preact` a lot here: [deno-lume-preact-example/tree/jsx-auto-import](https://github.com/jrson83/deno-lume-preact-example/tree/jsx-auto-import).

For the plugin it makes sense, to provide `preact` as export/import, like the `React` plugin does, I think.

oscarotero commented on preactjsx/mod.ts:
```diff
@@ -0,0 +1,81 @@
+  extensions: [".jsx", ".tsx"],
+};
+
+window.preact ||= preact;
```
I think this is not neccesary. The jsx library is imported automaticaly thanks to the jsxImportSource configuration in deno.json

oscarotero commented:
Thanks! I've added a comment for a small change.

In addition to that, the pages `object.jsx` and `string.jsx` are not saved into `_site` because they don't have a `default` export, so they don't produce any content (and empty pages aren't saved). If their purpose is only being imported by `index.jsx`, I'd saved them in an subdirectory ignored by Lume (with the name `_utils`, for example, or ignored in the _config.ts with `site.ignore()`). This prevents to be loaded needlessly.

jrson83 commented:
> Thanks! I've added a comment for a small change.
> 
> In addition to that, the pages `object.jsx` and `string.jsx` are not saved into `_site` because they don't have a `default` export, so they don't produce any content (and empty pages aren't saved). If their purpose is only being imported by `index.jsx`, I'd saved them in an subdirectory ignored by Lume (with the name `_utils`, for example, or ignored in the _config.ts with `site.ignore()`). This prevents to be loaded needlessly.

Yeah, right, I was mixing things up.

I resolved the issues:
- Added `?deps=preact@10.10.0` to `preact-render-to-string` import, since `react-render-to-string@5.2.1` is still pointing to `preact@10.9.0` (seems they did not update to the correct import for the esm.sh package)
- Rewritten `deps.ts` exports & `mod.ts` imports for `preact`
- Renamed `preact`  import in `_config.ts` to `preactjsx`, to not interfer with actual preact var
- Removed the `object.jsx` & `string.jsx` since they were just for internal testing purpose

oscarotero commented:
Great work.
Merged, thanks!
